File: h6/AirPlane.rb
class AirPlane

@@seats = Array.new(6) {Array.new(27)}

def initialize
	i = 0; c = 0
	while (i < 6 and c < 27)
		@@seats[i][c] = 0
		i += 1
		if(i == 6)
			c += 1; i = 0
		end
	end
end

def printSeats
	i = 0; c = 0
	while (i < 6 and c < 27)
		print @@seats[i][c]
		i += 1                        
		if(i == 6)
			c += 1; i = 0
			printf "\n"
		end
		if(i == 3)
			print "  "
		end
	end
end

def add(value)
	bool = [true, false]
	free = false
	i = 0; c = 0; flag = 0
	while(free == false and i < 27)
		if(@@seats[c][i] == 0)
			if(c+value <= 6)
				t = 0
				while(t < value)
					if(@@seats[c+t][i] != 0)
						i += 1; c = 0
						break;
					else
						free = true
						if(c < 3)
							flag = 1
						end
					end
					t += 1
				end
			else
				i += 1; c = 0;
				if(i >= 27)
					break;
				end
			end
		else
			c += 1
			if(c == 6)
				i += 1; c = 0
			end
			if(i > 27)
				break;
			end
		end
	end
	if(flag == 1 and c+value > 3)
		c += (3 - c)
	end
	t = 0
	while(t < value)
		@@seats[c+t][i] = value;
		t += 1
	end
end

def freeSeats
	i = 0; c = 0; counter = 0
	while(i < 27)
		while(c < 6)
			if(@@seats[c][i] == 0)
				counter += 1
			end
			c += 1
		end
		i += 1; c = 0
	end
	return counter
end

end


plane = AirPlane.new
printf "Free seats before : "
print plane.freeSeats
printf "\n"
	while(plane.freeSeats != 0)
		randNum = rand(3) + 1
		plane.add(randNum)
	end
plane.printSeats
printf "Free seats after : "
print plane.freeSeats
printf "\n"

File: h6/AirPlane.java
/*
Училище - Технологично училище "Електронни системи" към ТУ-София{ www.elsys-bg.org }
Клас - 11Б клас
Номер - 3
Име - Анна-Мария Верева Матейна

Задача - Да се направи програма имитираща местата в самолет - 27 реда, 2 колони[по 3 места в колона]. Де се подреждт пътниците в самолета, докато той не се напълни, и ако са повече от един да не бъдат разделяни. 
*/

import java.util.Random;

public class AirPlane{

	private int[][] seats = new int[6][27];
	private boolean full = false;
public void PrintSeats(){
	for(int c = 0; c < 27; c++){
		for(int n = 0; n < 6; n++){
			System.out.print(seats[n][c]);
			if(n == 2){
				System.out.print(" ");
			}
		}
		System.out.println();
	}
}
public void Add(int value){
	for(int r = 0; r < 27; r++){
		for(int k = 0; k < 6; k++){
			if(seats[k][r] == 0){
				if(value == 1){
					seats[k][r] = 1;
					full = true;
					break;
				}
						
				if(value == 2 && k<=4){
					if(seats[k+1][r] == 0 && k%3 <=1){
						seats[k][r] = 2;
						seats[k+1][r] = 2;
						full = true;
						break;
					}
				}
						
				if(value == 3 && k<=3){
					if(seats[k+1][r] == 0 && seats[k+2][r] == 0 && k%3 == 0){
						seats[k][r] = 3;
						seats[k+1][r] = 3;
						seats[k+2][r] = 3;
						f=true;
						break;
					}
				}
			}else{
				continue;
			}
		}
		if(full == true){
			break;
		}
	}
	if(full == false){
		System.out.println("Stop");
	}
}
public int FreeSeats(){
	int c = 0;
	for(int r = 0; r < 27; r++){
		for(int k = 0; k < 6; k++){
			if(seats[r][k] == 0){
				c++ ;
			}
		}
	}
	return c;
}
public static void main(String args[]){
	AirPlane plane = new AirPlane();	
	plane.PrintSeats();
	System.out.println("#######");
		while(plane.FreeSeats() != 0){
			int value = new Random().nextInt(3)+1;
			plane.Add(value);
		}
	plane.PrintSeats();
	}
}


